src/gs-discovery-feed-content-provider.c

```diff
@@ -136,6 +130,38 @@ get_app_thumbnail_cached_filename (GsApp *app)
 	return NULL;
 }
 
+static gboolean
+filter_for_discovery_feed_apps (GsApp *app, gpointer user_data)
+{
+	g_autofree gchar *app_thumbnail_filename = NULL;
+	GdkPixbuf *pixbuf;
+	g_autoptr(GVariant) icon_serialized = NULL;
+
+	if (gs_app_get_state (app) != AS_APP_STATE_AVAILABLE)
+		return FALSE;
+
+	/* This app must have an icon which is serializable */
+	pixbuf = gs_app_get_pixbuf (app);
+	if (pixbuf != NULL)
+		icon_serialized = g_icon_serialize (G_ICON (pixbuf));
```

Comment:
Serialization can be a pretty expensive operation, and it is very unlikely that it will fail. I would lean towards assuming that if `gs_app_get_pixbuf()` returns non-NULL, it will succeed, and so perhaps only add a warning if serialization fails later when building the variant.

Related: since you search specifically for apps that have an icon (https://github.com/endlessm/gnome-software/blob/e46ff4fb7841008b94cde0394c2b6991a59d944f/src/gs-discovery-feed-content-provider.c#L248), do we need the pixbuf check here?

Reply:
Yeah, I did notice that.

I think as a defensive measure, we probably want to at least ensure that if an app does turn out to have a null pixbuf / is unserializable that we just emit a warning and move on.